fastparquet/dataframe.py

```diff
@@ -153,9 +142,15 @@ def cat(col):
             else:
                 views[col] = block.values[i]
 
-    if index_names:
-        df.index.names = [
-            None if re.match(r'__index_level_\d+__', n) else n
-            for n in index_names
-        ]
     return df, views
+    # return df, indexes, views
+
+
+def finalize(df, indexes):
+    n = len(indexes)
+    if n == 1:
+        df.index = indexes[0]
```

Comment:
This shouldn't be necessary, the n==1 case is already special and dealt with directly in `empty()`

Reply:
This case does not happen with the dask compat commit